### DISCUSSION TITLE
Adding absolute or relative path of file to statusline.

### DISCUSSION THREAD

**Discussion by magnusriga** [sxyazi/yazi#1915] [1. Q&A]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

Wezterm

### `yazi --debug` output

```shell
Yazi
    Version: 0.3.3 (VERGEN_IDEMPOTENT_OUTPUT 2024-11-10)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (VERGEN_IDEMPOTENT_OUTPUT 2024-11-10)

Emulator
    Emulator.via_env: ("xterm-256color", "WezTerm")
    Emulator.via_csi: Ok(Unknown([]))
    Emulator.detect : WezTerm

Adapter
    Adapter.matches: Iterm2

Desktop
    XDG_SESSION_TYPE           : None
    WAYLAND_DISPLAY            : Some("wayland-0")
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: true

Variables
    SHELL              : Some("/usr/bin/zsh")
    EDITOR             : Some("nvim")
    VISUAL             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : tmux 3.5a
    ZELLIJ_SESSION_NAME: None
    Zellij version     : Permission denied (os error 13)

Dependencies
    file             : 5.41
    ueberzugpp       : Permission denied (os error 13)
    ffmpegthumbnailer: 2.2.3
    magick           : 7.1.1-41
    fzf              : 0.56.3
    fd               : 8.3.1
    rg               : 13.0.0
    chafa            : Permission denied (os error 13)
    zoxide           : 0.9.6
    7z               : Permission denied (os error 13)
    7zz              : 24.08
    jq               : 1.6
```

### Describe the question

Thanks for an awesome program.

**Question**: When browsing with yazi, I currently do not see which folder I am in. Example:

<img width="454" alt="image" src="https://github.com/user-attachments/assets/b1c0f66a-54f7-47e1-958a-d0cf02402980">

It would be great if the actual path to the folder I am in would be showed somewhere. For instance, as part of the file name in the status line. Either relative (from pwd) or absolute, would work well.

Is that possible?

Thanks again.

### Validations

- [X] I have searched the existing discussions/issues

**Comment by AnirudhG07** [comment]
Add this to your main `init.lua` for yazi, and you should get the `absolute path`
```lua
function Status:name()
	local h = self._tab.current.hovered
	if not h then
		return ui.Line({})
	end

	return ui.Line(" " .. tostring(h.url))
end
```

Hope it solves your issue :)

**Reply by AnirudhG07** [reply]
Also what you sent is not `yazi --debug` but the log file... so for any future issues/discussions, pls share the correct `yazi --debug`. :>

**Reply by magnusriga** [reply]
Thank you @AnirudhG07, very helpful. Any idea why the below snippet, i.e. with [file.prefix()](https://yazi-rs.github.io/docs/plugins/types/#app-data.folder-file) does not work (no statusline is shown)?

```lua
function Status:name()
  local h = self._tab.current.hovered
  if not h then
    return ui.Line({})
  end
  local prefix = h.prefix()
  local linked = ""
  if h.link_to ~= nil then
    linked = " -> " .. tostring(h.link_to)
  end
  return ui.Line(" " .. prefix .. h.name .. linked)
end
```

I am trying to turn the absolute paths into paths relative to CWD, or at the very least swap `/home/username` with `~`.

**Reply by AnirudhG07** [reply]
```lua
function Status:name()
	local h = cx.active.current.hovered
	if not h then
		return ui.Span("")
	end

	local linked = h.link_to and (" -> " .. tostring(h.link_to)) or ""
	local path = tostring(h.url)
	local prefix = ya.target_os() == "macos" and path:match("^(/Users/[^/]+/)")
		or ya.target_os() == "linux" and path:match("^(/home/[^/]+/)")
		or ""
	local name = path:find(prefix, 1, true) and path:sub(#prefix + 1) or path

	return ui.Span(" " .. name .. linked)
end
```
This is a work around for what you wanted. This is like manual changes to the name. But it removed the `/home/username` atleast. Hope this is good. I am not sure about `file.prefix()` one either.

**Reply by sxyazi** [reply]
> Any idea why the below snippet, i.e. with [file.prefix()](https://yazi-rs.github.io/docs/plugins/types/#app-data.folder-file) does not work

`prefix` is a method not a function, so you need `h:prefix()` not `h.prefix()`

**Reply by AnirudhG07** [reply]
Right. Lol.

**Comment by magnusriga** [comment]
Thank you both

The below works, but is it not possible to use envs in the prefix string? I could not make `/home/$USERNAME` work.

Thank you.

```lua
 function Status:name()
  local h = self._tab.current.hovered
  local url = h.url
  local home_prefix = "/home/magnus"
  if not h then
    return ui.Line({})
  end
  if url:starts_with(home_prefix) then
    url = Url("~/"):join(url:strip_prefix(home_prefix))
  end
  local linked = ""
  if h.link_to ~= nil then
    linked = " -> " .. tostring(h.link_to)
  end
  return ui.Line(" " .. url .. linked)
end
````

**Reply by AnirudhG07** [reply]
Sorry I didnt get you. Can you explain a bit more???

**Reply by magnusriga** [reply]
If I write `local home_prefix = "/home/$USERNAME"`, will yazi expand the `$USERNAME` environment variable (which exists in my shell environment), like is done in normal shell parameter substitution? Or do I have to hardcode in the username (`magnus`)?

**Reply by AnirudhG07** [reply]
you can use `ya.user_name()`. So your code looks like - 
```lua
 function Status:name()
  local h = self._tab.current.hovered
  local url = h.url
  local home_prefix = "/home/" .. tostring(ya.user_name())
  if not h then
    return ui.Line({})
  end
  if url:starts_with(home_prefix) then
    url = Url("~/"):join(url:strip_prefix(home_prefix))
  end
  local linked = ""
  if h.link_to ~= nil then
    linked = " -> " .. tostring(h.link_to)
  end
  return ui.Line(" " .. url .. linked)
end
```
